tests/test_mlflow_integration.py

```diff
@@ -0,0 +1,154 @@
+        assert False, f"Unhandled exception while loading a model: {e}"
+
+    try:
+        # list_artifacts will use the UC OSS model URI and make REST API calls to
+        # UC OSS to:
+        #   1) retrieve credentials (none for file based UC OSS)
+        #   2) use the storage location returned from UC OSS for the model version
+        #      list the artifacts stored in the location
+        mlflow.artifacts.list_artifacts(model_uri)
+    except Exception as e:
+        assert False, f"Unhandled exception while listing artifacts: {e}"
+
+    path = os.path.join("/tmp", "models", model_name, str(model_version))
+
+    try:
+        # download_artifacts will use the UC OSS model URI and make REST API calls
+        # to UC OSS to:
+        #   1) retrieve credentials (none for file based UC OSS)
+        #   2) copy the artifact files from the storage location to the
+        #      destination path
+        mlflow.artifacts.download_artifacts(
+            artifact_uri=f"models:/{model_name}/{model_version}",
+            dst_path=path,
+        )
+        requirements_path = f"{path}/requirements.txt"
+        assert os.path.exists(requirements_path), f"File {requirements_path} does not exist."
+        with open(requirements_path) as file:
+            lines = file.readlines()
+        assert len(lines) > 0
+    except Exception as e:
+        assert False, f"Unhandled exception while testing download_artifacts: {e}"
+
+    # Test get RM/MV works
+    model1 = mlflow.MlflowClient().get_registered_model(model_name)
+    assert model1.name == model_name
+    assert model1.description == ""
+    model_v1 = mlflow.MlflowClient().get_model_version(name=model_name, version=model_version)
+    assert model_v1.name == model_name
+    assert model_v1.version == 1
+    assert model_v1.description == ""
+
+    # Test update RM/MV works
+    mlflow.MlflowClient().update_registered_model(model_name, description=rm_desc)
+    model2 = mlflow.MlflowClient().get_registered_model(model_name)
+    assert model2.name == model_name
+    assert model2.description == rm_desc
+    mlflow.MlflowClient().update_model_version(
+        name=model_name, version=model_version, description=mv_desc
+    )
+    model_v1_2 = mlflow.MlflowClient().get_model_version(name=model_name, version=model_version)
+    assert model_v1_2.name == model_name
+    assert model_v1_2.version == 1
+    assert model_v1_2.description == mv_desc
+
+    rms = mlflow.MlflowClient().search_registered_models()
+    assert len(rms) == 1
+    mvs = mlflow.MlflowClient().search_model_versions(f"name='{model_name}'")
+    assert len(mvs) == 1
+    mlflow.MlflowClient().delete_model_version(name=model_name, version=1)
+    mvs = mlflow.MlflowClient().search_model_versions(f"name='{model_name}'")
+    assert len(mvs) == 0
+    mlflow.MlflowClient().delete_registered_model(name=model_name)
+    rms = mlflow.MlflowClient().search_registered_models()
+    assert len(rms) == 0
+    try:
```

Comment:
This can be converted to a with pytest.raises() call to ensure that the model version is not present any longer